js_modules/dagit/packages/core/src/ticks/SensorDryRun.tsx

```diff
@@ -0,0 +1,361 @@
+import {assertUnreachable} from '../app/Util';
+import {PythonErrorFragment} from '../app/types/PythonErrorFragment.types';
+import {SET_CURSOR_MUTATION} from '../sensors/EditCursorDialog';
+import {
+  SetSensorCursorMutation,
+  SetSensorCursorMutationVariables,
+} from '../sensors/types/EditCursorDialog.types';
+import {testId} from '../testing/testId';
+import {RepoAddress} from '../workspace/types';
+
+import {RunRequestTable} from './DryRunRequestTable';
+import {SensorDryRunMutation, SensorDryRunMutationVariables} from './types/SensorDryRun.types';
+
+type DryRunInstigationTick = Extract<
+  SensorDryRunMutation['sensorDryRun'],
+  {__typename: 'DryRunInstigationTick'}
+>;
+
+export const SensorDryRun: React.FC<{
+  name: string;
+  onClose: () => void;
+  repoAddress: RepoAddress;
+  currentCursor: string;
+  isOpen: boolean;
+  jobName: string;
+}> = ({repoAddress, name, currentCursor, onClose, jobName, isOpen}) => {
+  const [sensorDryRun] = useMutation<SensorDryRunMutation, SensorDryRunMutationVariables>(
+    EVALUATE_SENSOR_MUTATION,
+  );
+
+  const [cursor, setCursor] = React.useState(currentCursor);
+
+  const [submitting, setSubmitting] = React.useState(false);
+  const [error, setError] = React.useState<PythonErrorFragment | null>(null);
+  const [
+    sensorExecutionData,
+    setSensorExecutionData,
+  ] = React.useState<DryRunInstigationTick | null>(null);
+
+  const sensorSelector = React.useMemo(
+    () => ({
+      sensorName: name,
+      repositoryLocationName: repoAddress.location,
+      repositoryName: repoAddress.name,
+    }),
+    [repoAddress, name],
+  );
+
+  const submitTest = React.useCallback(async () => {
+    debugger;
+    setSubmitting(true);
+    const result = await sensorDryRun({
+      variables: {
+        selectorData: sensorSelector,
+        cursor,
+      },
+    });
+    const data = result.data?.sensorDryRun;
+    if (data) {
+      if (data?.__typename === 'DryRunInstigationTick') {
+        if (data.evaluationResult?.error) {
+          setError(data.evaluationResult.error);
+        } else {
+          setSensorExecutionData(data);
+        }
+      } else if (data?.__typename === 'SensorNotFoundError') {
+        showCustomAlert({
+          title: 'Sensor not found',
+          body: `Could not find a sensor named: ${name}`,
+        });
+      } else {
+        setError(data);
+      }
+    } else {
+      assertUnreachable('sensorDryRun Mutation returned no data??' as never);
+    }
+    setSubmitting(false);
+  }, [sensorDryRun, sensorSelector, cursor, name]);
+
+  console.log(repoAddress);
+
+  const buttons = React.useMemo(() => {
+    if (sensorExecutionData || error) {
+      return (
+        <Button intent="primary" onClick={onClose}>
+          Close
+        </Button>
+      );
+    }
+    if (submitting) {
+      return (
+        <Box flex={{direction: 'row', gap: 8}}>
+          <Button onClick={onClose}>Cancel</Button>
+        </Box>
+      );
```

Comment:
Is it possible to cancel while it's submitting? Does the "Submit" button get hidden during the "submitting" phase?